### DISCUSSION TITLE
Stiffness Tensor Terms for Springs

### DISCUSSION THREAD

**Discussion by jmcguire8** [OpenRadioss/OpenRadioss#2656] [Contributors]
For springs in OpenRadioss, Type 8 or Type 13, it would seem that we can only define the diagonal terms of the elasticity tensor (stiffness is associated with 6 independent DOFs). Is this true? Or if not, can you help me understand how I may be able to define (more of) the 21 terms in the stiffness tensor? Are there options for a connector in OpenRadioss that have more stiffness terms accessible? I am looking for something similar to Abaqus linear uncoupled elastic behavior for connectors.

![image](https://github.com/OpenRadioss/OpenRadioss/assets/142855615/ff65eadf-a5df-4351-91fc-77ed6d8eae18)

**Comment by MarianBulla** [comment]
Hello @jmcguire8,
thank you for your interest in OpenRadioss.
From what I can see, you should go with a full solid element.
Please have a look on YouTube, how I'm craeting a USER material using the Hook's law, for an isotropic material behavior, where only few of these stiffness values are used. 

https://www.youtube.com/watch?v=Js_i7CnZty0&t=654s 
At time ~13:15 
You can find everything on the OpenRadioss Tools repository: https://github.com/OpenRadioss/Tools/tree/main/userlib_sdk/examples/USER_material_laws_for_solids_and_shells/Solids

For your application, the material model is much more complex, but doable.

Hope it helps you ?

Thanks and best regards,
Marian

**Reply by jmcguire8** [reply]
Thank you @MarianBulla for the details.

The appeal of the connector in Abaqus is that I can approximate the stiffness of a system of components at a single point. A part is coupled to the connector. As the part is loaded, I can measure the forces, moments, displacement, and rotations at the connector. So this approach accomplishes 3 things:
1. Simplify an assembly stiffness for computational efficiency
2. Apply this stiffness at a specific point of interest
3. Measure forces, moments, etc. from this point of interest resisting the motion of a connected part

It's not immediately clear to me how I can use a solid element in a similar manner.
